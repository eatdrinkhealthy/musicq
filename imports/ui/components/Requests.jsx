// @flow
import React, {
  Component,
} from "react";
import type { IRequest } from "../../data/sampleRequests";

type IRequestProps = {
  request: IRequest,
  handleSelectRequest: (nextReqUrl: string) => void,
};

export class Request extends Component {
  props: IRequestProps;

  handleSelectRequest() {
    this.props.handleSelectRequest("TODO get url");
  }

  render() {  // eslint-disable-line flowtype/require-return-type
    return (
      <div>
        <input
          type="button"
          className="request"
          onClick={this.handleSelectRequest}
          value={`${this.props.request.artist} - ${this.props.request.title}`}
        />
      </div>
    );
  }
}

type IRequestsProps = {
  requests: Array<IRequest>,
  handleSelectRequest: (nextReqUrl: string) => void,
};

const Requests = (props: IRequestsProps): React$Element<*> => (
  <div>
    <h4>Requests</h4>
    <ul>
      {props.requests.map((request: IRequest): React$Element<*> =>
        (<Request
          key={request.id}
          request={request}
          handleSelectRequest={props.handleSelectRequest}
        />))
      }
    </ul>
  </div>
);

export default Requests;
