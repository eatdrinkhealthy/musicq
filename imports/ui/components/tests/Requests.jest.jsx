// @flow
/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
/* eslint-disable import/no-extraneous-dependencies */

import React from "react";
import renderer from "react-test-renderer";
import Requests, { Request } from "../Requests";

const sampleRequests = [
  {
    id: "1",
    title: "Jeep's Blues",
    artist: "Duke Ellington",
    songUrl: "https://www.youtube.com/watch?v=uUcEGOLfUTE",
    submittedBy: "smarsh",
  },
  {
    id: "2",
    title: "Ya Mar",
    artist: "The Mustangs",
    songUrl: "https://www.youtube.com/watch?v=uwHR7Pji2cg",
    submittedBy: "smarsh",
  },
  {
    id: "3",
    title: "Gone",
    artist: "The Black Crowes",
    songUrl: "https://www.youtube.com/watch?v=31tXHtPUCUk",
    submittedBy: "smarsh",
  },
  {
    id: "4",
    title: "Franklin's Tower",
    artist: "Grateful Dead",
    songUrl: "https://www.youtube.com/watch?v=6J99HEY67hc",
    submittedBy: "smarsh",
  },
];

describe("<Request />", function () {
  it("matches render snapshot", function () {
    const handleSelectRequest = jest.fn();
    const tree = renderer.create(<Request
      request={sampleRequests[0]}
      nowPlaying={false}
      handleSelectRequest={handleSelectRequest}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<Requests />", function () {
  it("matches render snapshot", function () {
    const handleSelectRequest = jest.fn();
    const tree = renderer.create(<Requests
      requests={sampleRequests}
      handleSelectRequest={handleSelectRequest}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
