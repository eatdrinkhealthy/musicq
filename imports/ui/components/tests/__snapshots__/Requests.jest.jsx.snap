exports[`<Request /> matches render snapshot 1`] = `
<div
  onClick={[Function]}>
  <span>
    Duke Ellington
     - 
  </span>
  <span>
    Jeep\'s Blues
  </span>
</div>
`;

exports[`<Requests /> matches render snapshot 1`] = `
<div>
  <h4>
    Requests
  </h4>
  <ul>
    <div
      onClick={[Function]}>
      <span>
        Duke Ellington
         - 
      </span>
      <span>
        Jeep\'s Blues
      </span>
    </div>
    <div
      onClick={[Function]}>
      <span>
        The Mustangs
         - 
      </span>
      <span>
        Ya Mar
      </span>
    </div>
    <div
      onClick={[Function]}>
      <span>
        The Black Crowes
         - 
      </span>
      <span>
        Gone
      </span>
    </div>
    <div
      onClick={[Function]}>
      <span>
        Grateful Dead
         - 
      </span>
      <span>
        Franklin\'s Tower
      </span>
    </div>
  </ul>
</div>
`;
